### ISSUE TITLE
Give Swarmer the values of Cyclones ammo multipliers or higher

### ISSUE THREAD

**Issue by the-red-soul** [Anuken/Mindustry-Suggestions#1880]
**Describe what you would like changed, and why.**

Cyclones ammo multipliers were recently buffed *at least in terms of surge alloy* to 5x, whereas swarmer was completely untouched with a 2x ammo multiplier for Surge and pyratite. 

Swarmer in its current state is LITERALLY UNUSABLE as a turret due to the immense ammo costs/productions necessary to keep even a very small amount of them firing. I had gone over actual numbers back in my last suggestion which I will link here. #1790

In version 5.0 swarmer was almost never used. In 6.0 swarmer is literally never used in any match I join. Wether its a community server or steam game. Could that just be coincidence or timing? maybe. The fact is is that swarmer is in dire need of balance changes to even make it remotely viable for use. 

**Describe the changes you want to propose. Include possible alternatives.**

Give swarmer ammo multipliers of either 5x or higher on all 3 ammo types. Or drastically increase damage output in order to compensate for the insane ammo usage.

**Comment by GreyMario** [comment]
Counterpoint to "literally unusable": you're using the wrong ammunition. Three cryo-cooled blast Swarmers against a Frozen target make quick work of T4s. Like, literally in single-digit seconds.

**Comment by the-red-soul** [comment]
and how much ammo did you use during so. Unreasonable amounts

**Comment by the-red-soul** [comment]
Ight let me rephrase my response. 

You were using BLAST compound which already has a 4x multiplier. The other two still have a 2x multiplier. 3 cryo cooled swarmers firing for single digit seconds, so at most 9 seconds. Swarmers firerate is 7.74 shots/sec. with cryo it is 14.706 shots/sec. Divide by four for the ammo multiplier of blast and you get 3.67 blast/sec for one turret. With three that is 11.03 blast/sec. for at most 9 seconds that equates to 93 blast in total.

One blast mixer makes one blast every 1.33 seconds, meaning if you have 3 cryo swarmers you need at least 9 blast factories in order for them to have continuous sustained fire. 

FOR THREE TURRETS???

**Comment by itcannotbe** [comment]
Is this a rant or is someone not familiar with the concept of storing ammo for turrets outside of the turrets

**Comment by the-red-soul** [comment]
1000 resources for a vault. 300 for a container. Takes how long to fill said vault due to the slow production speeds of pyratite, blast, and surge factories. 3 turrets with cryo use 11 blast/sec and 22 surge/pyratite per second. Three turrets. Why bother using swarmer when many other ammo efficient options exist in the form of surge cyclone, ripples, salvo, etc. 

Clearly you have never actively tried to use swarmers in a defence.

**Comment by itcannotbe** [comment]
And I suppose you're using 11 items per second, non-stop? Because?????????????????

**Comment by the-red-soul** [comment]
dont you think, its a bit rediculous that you need to feed 3 cryo swarmers more than one titanium belt can handle??

**Comment by itcannotbe** [comment]
> And I suppose you're using 11/22 items per second, non-stop? Because?????????????????

Answer the question.

**Comment by itcannotbe** [comment]
> dont you think, its a bit rediculous that you need to feed 3 cryo swarmers more than one titanium belt can handle??

Regardless of it might be tedious it may be but......
Wave spacing exist.
Unit Caps exist that prevents you from.getting swarmed non stop by enemy units.

**Comment by the-red-soul** [comment]
Because swarmers have to shoot things to kill them. And to shoot them they are required to have 3.76 blast or 7.52 pyratite/surge for one swarmer with cryo. And to kill a lot of units or units with high hp they need to keep shooting. Depending on your ammo distributution setup it is very easy to run out of ammunition in some of the other swarmers leading to more units surviving making other swarmers keep firing for a lot longer.

**Comment by itcannotbe** [comment]
> > And I suppose you're using 11/22 items per second, non-stop? Because?????????????????
> 
> Answer the question.

NON-STOP?

**Comment by itcannotbe** [comment]
![Screenshot_20210127_094441_io anuke mindustry](https://user-images.githubusercontent.com/45280919/105930429-9ebbf600-6084-11eb-9341-f83877a34ec2.jpg)
![Screenshot_20210127_094413_io anuke mindustry](https://user-images.githubusercontent.com/45280919/105930439-a2e81380-6084-11eb-952e-860da28d7a55.jpg)
![Screenshot_20210127_094400_io anuke mindustry](https://user-images.githubusercontent.com/45280919/105930449-a4b1d700-6084-11eb-8fdb-e270d7418c81.jpg)
![Screenshot_20210127_094349_io anuke mindustry](https://user-images.githubusercontent.com/45280919/105930455-a7143100-6084-11eb-9acd-fd10c9216c4c.jpg)
![Screenshot_20210127_094309_io anuke mindustry](https://user-images.githubusercontent.com/45280919/105930456-a7143100-6084-11eb-91bf-a47aafe5a90e.jpg)

**Comment by B00M3000** [comment]
Are you dumb or what, if the container storing the buffer cannot even fill between rounds because the swarmer uses so much when what the point ur trying to make.

**Comment by itcannotbe** [comment]
> Are you dumb or what, if the container storing the buffer cannot even fill between rounds because the swarmer uses so much. What the point ur trying to make.

Ah yes. Your point hinges on the what if.

What kind of wave spacing do you even have? 0 seconds?

Unit production.multiplier 100x?

**Comment by B00M3000** [comment]
Tell me, how many surge factories are needed for 1 titanium belt.

**Comment by itcannotbe** [comment]
Tell me, why on you keep insisting on using surge as ammo?

I would understand if it's a foreshadow. But for a swarmer, your better off using blast TBH.

Shock from surge isn't even that good when used in a swarmer.

**Comment by genNAowl** [comment]
https://github.com/Anuken/Mindustry-Suggestions/issues/1790

**Comment by genNAowl** [comment]
I think he means that as long as it is an ammo choice, it should be balanced.

**Comment by B00M3000** [comment]
> Tell me, why on you keep insisting on using surge as ammo? I would understand if it's a foreshadow. But for a swarmer, your better off using blast TBH. Shock from surge isn't even that good when used in a swarmer.

Why do you say that? Shock can dmg multiple troops while blast can only dmg 1. when u get to t4 and 5's

**Comment by B00M3000** [comment]
> I think he means that as long as it is an ammo choice, it should be balanced.

Thank you.

**Comment by DarkChemist** [comment]
First, swarmer shock effect is larger than cyclones, so it has a larger affected area.
Second, it is very pointless when you use a huge amount of surge/blast to defend. There is no reason for players creating so much surge or blast to feed just a few swarmers. Even though you have unlimited wave spacing, you need a huge surge production, like a full plast belt of surge/blast to feed 10-20 swarmers quickly. It is very hard for players to make full plast belt of surge/blast with current factories. Even though you have filled a lot of vaults with surge to feed them, why don't just use cyclones instead for better ammo consumption and damage? When Anuke decided to buff surge cyclones to 5x ammo multiplier, there is no point to use surge in swarmers. Even when you use blast in swarmers, u can just use them in cyclones or ripple for better damage and less ammo consumption. Current swarmers are very unbalanced and it is never used in a normal survival map gameplay. I think this turret needs a huge revamp in order to be balanced, like shooting big a missile a and creates an effect for example, surge for shock effect, in that area.

**Comment by itcannotbe** [comment]
Just to clarify:

Shock isn't aoe. It's like arc turret but bullet form.

**Comment by claw4oo** [comment]
In the current state of the swarmer, it is by far one of the worse turret options to use, relative to its cost.
Red soul is trying to push for it to be more balanced compared to other turrets

**Comment by GreyMario** [comment]
This conversation should have never happened because of the magical concept of _building up a reserve of ammunition_.

**Comment by the-red-soul** [comment]
im sorry let me build up an ammo reserve that gets depleted in around 9 seconds and then the turrets stop firing completely.

**Comment by RedstonerKBC** [comment]
Good luck finding enough ammo to fill your swarmers faster than they can shoot it all away. I've tested this too and it does take an insane amount of resources for what you get out of the swarmer. I was able to survive quite a long time with swarmers but they drain the ammo so quickly even with storage that come the next wave only half my swarmers were firing at the enemies. The damage they do for the expensive resources they require seems unbalanced in my opinion. This was in Build 122 so things might've changed, but it wasn't that long ago that I tried them to moderate success.

**Comment by itcannotbe** [comment]
> First, swarmer shock effect is larger than cyclones, so it has a larger affected area.

Wrong. See images above.

**Comment by itcannotbe** [comment]
> First, swarmer shock effect is larger than cyclones, so it has a larger affected area.

Wrong. See images above.

**Comment by the-red-soul** [comment]
Firstly that has nothing to do with the suggestion itself. Secondly Swarmer DOES have larger surge arcs, and if you looked at your images you had posted you would have clearly seen that Swarmers ammo usage in comparison to all the other turrets is exceptionally high with surge and pyratite ammo. Especially considering the production rates of the Pyratite Mixer and Alloy Smelter. Spectres ammo multiplier being two is not nearly as severe as swarmers due to one airblast drill on a 4x4 pad of thorium with water produces 6.46 thorium per second. Most of the other ammunition posted is a lot easier to produce in mass quantities than blast, pyratite and surge. 
![image](https://user-images.githubusercontent.com/31638088/105952490-def08980-602e-11eb-9b94-df325149fa17.png)

**Comment by Slava0135** [comment]
Swarmers need a buff

**Comment by itcannotbe** [comment]
Swarmers need a redesign of it's mechanic.

**Comment by B00M3000** [comment]
> Swarmer's need a redesign of it's mechanic.

Glad to see you have come to your senses.

**Comment by Slava0135** [comment]
> > Swarmer's need a redesign of it's mechanic.
> 
> Glad to see you have come to your senses.

I just want to nerf anything being overused (w A t E r E x T r A c T o r) and buff anything being underused.

**Comment by Jammo2000** [comment]
Communist detected. Should all things be equally used?

**Comment by B00M3000** [comment]
> Communist detected. Should all things be equally used?

no

**Comment by HTMLPhoton** [comment]
alright, I think we are stumbling upon a bigger issue. Which is why use swarmers over cyclones? They are both gated by plas, both have roughly the same resource costs (PER TILE). they play and feel roughly the same too - both are high-aoe rapid-fire turrets with roughly the same range

I think we need to apply Quant's law and nerf their weaknesses while buffing their strengths. and I think we should do that by increasing their ammo requirements, prehaps also increase the amount of ammo they can hold or let the player figure out how to supply them ammo, while buffing their damage output to be superior to cyclones.

If we instead just reduced their ammo consumption, we would just end up with a second cyclone.

**Comment by Slava0135** [comment]
Why not remove cyclone

**Comment by HTMLPhoton** [comment]
> Why not remove cyclone

I do not see how that is relevant to the conversation. The whole point of balance is so that things are viable. to completely remove anything would go against the very idea of balance

**Comment by Slava0135** [comment]
You pointed out that they are very similar. Then why do they both exist

**Comment by HTMLPhoton** [comment]
> You pointed out that they are very similar. Then why do they both exist

To say that is to say "Oh he is sick, let's let him die". you are treating the fact that they are very similar as unchanagable and set in stone, but that is not the case. they don't have to be similar, just like how a person doesn't have to be sick. the whole point of my idea was to make them not similar, analogous to curing the person so he is no longer sick.

**Comment by B00M3000** [comment]
I think that something that could give back the unique quality for the swarmer is to give back its infinite range or just very long range. Basically the ability to chase a player or unit down since no other turret can do such a thing.

**Comment by Jammo2000** [comment]
That sounds like an excellent way to make swimmer viable, my only worry is
logic control. One solution to that could be to make the bullets only seek
when they fire at something in range.

**Comment by itcannotbe** [comment]
Do swarmer bullets even re-target when the orginal target is dead?

**Comment by itcannotbe** [comment]
> That sounds like an excellent way to make swimmer viable, my only worry is logic control. One solution to that could be to make the bullets only seek when they fire at something in range.

When auto correct fucks with you

> swimmer

**Comment by B00M3000** [comment]
lol

**Comment by the-red-soul** [comment]
logic is already impossible to balance, if swarmer got its old range back it would be more viable yes, and more abuseable by logic, but it would justify the ammo usage I feel. TBH whats the difference between rangeboosted spectres and rangeboosted swarmers if they had their v5 range?

**Comment by SaltyBeggarBR** [comment]
> This conversation should have never happened because of the magical concept of _building up a reserve of ammunition_.

The problem though is output speed. If conveyors could instantly transport items from somewhere to else where that wouldn't be that much of a problem but because of the limited speed of conveyors and transporting items in general it ends up being a problem.

**Comment by RedstonerKBC** [comment]
Mass drivers don't need to worry about transporting items long distance or getting the way. Sure they're not as fast but there's nothing stopping you from placing multiple drivers.
